### PR TITLE
Fix #24830 Bring Back White Label Color Selectors

### LINKED ISSUE #24830: Bring Back White Label Color Selectors
### Problem Statement
Currently, the ability for the user to select primary and secondary colors to white label the dotCMS backoffice is not available. 

### Steps to Reproduce
1. Log in to the dotCMS > Settings.
2. Change the primary and secondary color
3. Nothing happens

### Acceptance Criteria
1. Restore the CSS property and the JavaScript that provide the ability to select primary and secondary colors to white label the dotCMS backoffice.
2. Ensure that the colors are updated throughout the entire backoffice. 

### External Links
N/A

### Assumptions & Initiation Needs
- Assumption: The code to provide the color selection functionality should still exist. 
- Initiation Need: The code must be reviewed and updated to ensure it still works as intended. 

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by fmontes**
### Summary
🎨🧹🛠️

This pull request updates the color scheme of the dotCMS UI and web components to use CSS variables and the PrimeNG button component. It changes the main color from purple to blue and the secondary color from violet to dark purple, and applies these colors consistently across the application. It also removes redundant CSS variables and improves the readability and maintainability of the SCSS and HTML files.

> _`CSS` variables are the new law_
> _We change the colors of them all_
> _From purple shades to blue and violet_
> _We make the UI look like a riot_

### Walkthrough
*  Update the CSS variables for the root element to use the new color scheme for the dotCMS UI in the following files: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-7f6b5e33c4ea41947a136089cf7e0e4622468edd67ba3f172d34d04241d0b528L44-R51), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-0e5ff256cb530748db0d3b90134763141495c6a256e24c8e9ae7081eb19f4035L37-R41), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-7fa115edf7f56fb7bb2d6516c32abf8ee398bfe192708be431479877c26aeff1L11-R15), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-e9f94cec3297e58f16509df1f61391406e327cf1622cb6dbb945e98ff627334eL11-R18), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-341d7c5bc51d9d0ddde184ff8a0440138df8065d32811c5505de8f08dbbd847fL142-R150))
*  Change the link to request a license to use the PrimeNG button component and remove the unnecessary CSS rules in the DotUnlicenseComponent: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-ac97712467059b55a407b570a82eb281486e334ee7c9ca7da4c3b13fd10062f3L20-R20), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-c1973854ef2bf60bbe380b218977c9c0ff18193ff3fe96b15a77a854e058e3b0L51-L67), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-aeb88de09390077bfa908b54943b053cfb9e68f93bec22bbf4d58124643455f8L2-R10))
*  Remove the text-decoration property and use the CSS variables for the main color in the CSS rules for the PrimeNG button component in the dotcms-theme module: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-6f8b74847e0ed7761e15eb27e3a92e988f2d91e52df19923bbb1177913c71187R4), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-6f8b74847e0ed7761e15eb27e3a92e988f2d91e52df19923bbb1177913c71187L16-R20))
*  Remove the duplicated CSS variables for the root element from the styles.scss and dotcms.scss files: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-19fbe1ab2e0dcad6949b44c72556e4893c233c54263211aa055d0034241fcb5fL9-L21), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-32778be547d7c5e6d0e5d99589677990b7aa3f58561372b96d2d33d28e1f5c18L25-L34))
*  Use the CSS variables for the main and secondary colors instead of the hardcoded values in the SCSS variables for the primary and secondary color palettes: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-a673541496e37f0c8d73eef81a2026235e3d621daf64af8bf35a97881cb2dce5L8-R9), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-a673541496e37f0c8d73eef81a2026235e3d621daf64af8bf35a97881cb2dce5L19-R19))
*  Format the link tags in the preview-head.html file for the dotCMS web components: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-e9f94cec3297e58f16509df1f61391406e327cf1622cb6dbb945e98ff627334eL2-R7))
*  Update the border, background, arrow and text colors of various Dojo components to use the new main color instead of the old one in the dotcms.css file: ([link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3765-R3765), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3772-R3775), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3782-R3785), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3793-R3793), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3803-R3803), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3809-R3810), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3820-R3820), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3865-R3866), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3881-R3885), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3930-R3942), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL3951-R3976), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4015-R4016), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4043-R4044), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4055-R4062), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4073-R4074), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4092-R4092), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4099-R4100), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4114-R4114), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4121-R4122), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4143-R4144), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4172-R4173), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4188-R4188), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4201-R4202), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4216-R4216), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4223-R4224), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4238-R4238), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4245-R4246), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4260-R4260), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4290-R4291), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4302-R4309), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4327-R4328), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4347-R4352), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4413-R4417), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4428-R4428), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4435-R4439), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4450-R4450), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4493-R4500), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4507-R4519), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4548-R4549), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4570-R4570), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4593-R4593), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4601-R4601), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4637-R4643), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4663-R4663), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL4939-R4939), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL5036-R5042), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL5334-R5334), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL5477-R5477), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6154-R6154), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6161-R6161), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6312-R6318), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6331-R6332), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6350-R6350), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6385-R6385), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6400-R6400), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL6724-R6724), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL9131-R9131), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL9174-R9174), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL9382-R9382), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL9921-R9921), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL10055-R10055), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL10090-R10090), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL10349-R10349), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL10388-R10388), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL11025-R11025), [link](https://github.com/dotCMS/core/pull/24876/files?diff=unified&w=0#diff-08d797a1a8ebab0a7a24680f4640b5434967d4e29c17783a873ad53dfefad1cdL11590-L11598))

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/api/services/dot-ui-colors/dot-ui-colors.service.ts [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
+        return {
+            hue: match[1],
+            saturation: match[2],
+            lightness: match[3]
+        };
+    } else {
+        // Handle case where input was not in correct format
+        throw new Error('Input is not a valid HSL color string');
+    }
+}
```
Comment (zJaaal) [inline core-web/apps/dotcms-ui/src/app/api/services/dot-ui-colors/dot-ui-colors.service.ts:39]: maybe you can also check the length of the matches, just to be sure

File: core-web/libs/dotcms-webcomponents/stencil.config.ts [hunk 2]
```diff
@@ -47,5 +47,10 @@ export const config: Config = {
         ]
     },
 
-    plugins: [sass()]
+    plugins: [
+        sass({
+            // injectGlobalPaths: [`${__dirname}/dotcms-scss/shared/_colors.scss`],
+            // includePaths: [`/libs/dotcms-scss/shared`],
```
Comment (zJaaal) [inline core-web/libs/dotcms-webcomponents/stencil.config.ts:53]: do we need this comments?

**Review by zJaaal — APPROVED** [review]
Do we need to compile the block editor after this changes?

**Review by danielmdob — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Comment by fmontes** [comment]
> Do we need to compile the block editor after this changes?

Yes! excelent catch.

**Pull request merged**